src/controllers/userAddress.js

```diff
@@ -0,0 +1,33 @@
+require('dotenv').config()
+const User = require('../../models/User')
+const fetch = require('node-fetch')
+const { StatusCodes } = require('http-status-codes')
+
+const getAddressCoordinate = async (req, res) => {
+    const address = req.query.address
+    const GEOAPIFY_API_KEY = process.env.GEOAPIFY_API_KEY
+    const url = `https://api.geoapify.com/v1/geocode/search?text=${address}&apiKey=${GEOAPIFY_API_KEY}`
+
+    try {
+        const result = await fetch(url)
+        const data = await result.json()
+        res.status(StatusCodes.OK).json(data)
+    } catch (err) {
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json(err.message)
+    }
+}
+
+const getAddressAutocomplete = async (req, res) => {
+    const address = req.query.address
+    const GEOAPIFY_API_KEY = process.env.GEOAPIFY_API_KEY
+    const url = `https://api.geoapify.com/v1/geocode/autocomplete?text=${address}&apiKey=${GEOAPIFY_API_KEY}`
+
+    try {
+        const result = await fetch(url)
+        const data = await result.json()
```

Comment:
Let's simplify the data we return from this to the following keys: formatted, lat, lon. You would just need to map through the features array and pull out this information. Can you do the same for getAddressCoordinate as well? The only difference with getAddressCoordinate is that it should just return the top result rather than an array of suggestions.
![image](https://user-images.githubusercontent.com/70294155/228699103-0b71555d-5edd-4e43-b4f4-54a7814b204d.png)

Reply:
When we return data to our end user let's use latitude and longitude rather than the lat/lon abbreviations to keep consistent with the rest of the app.